File: src/Stronghold.Application/Features/UserMemberships/UserMembershipResponse.cs
namespace Stronghold.Application.Features.UserMemberships;

public class UserMembershipResponse
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public string UserFullName { get; set; } = string.Empty;
    public int MembershipPackageId { get; set; }
    public string MembershipPackageName { get; set; } = string.Empty;
    public decimal MembershipPackagePrice { get; set; }
    public DateTime StartDate { get; set; }
    public DateTime EndDate { get; set; }
    public bool IsActive { get; set; }
    public DateTime CreatedAt { get; set; }
}

File: stronghold_desktop/lib/features/memberships/models/user_membership_response.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'user_membership_response.freezed.dart';
part 'user_membership_response.g.dart';

@freezed
abstract class UserMembershipResponse with _$UserMembershipResponse {
  const factory UserMembershipResponse({
    required int id,
    required int userId,
    required String userFullName,
    required int membershipPackageId,
    required String membershipPackageName,
    required double membershipPackagePrice,
    required DateTime startDate,
    required DateTime endDate,
    required bool isActive,
    required DateTime createdAt,
  }) = _UserMembershipResponse;

  factory UserMembershipResponse.fromJson(Map<String, dynamic> json) =>
      _$UserMembershipResponseFromJson(json);
}

@freezed
abstract class PagedUserMembershipResponse with _$PagedUserMembershipResponse {
  const factory PagedUserMembershipResponse({
    required List<UserMembershipResponse> items,
    required int totalCount,
    required int totalPages,
    required int currentPage,
    required int pageSize,
  }) = _PagedUserMembershipResponse;

  factory PagedUserMembershipResponse.fromJson(Map<String, dynamic> json) =>
      _$PagedUserMembershipResponseFromJson(json);
}
